pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -271,7 +272,9 @@ func checkIfDaemonPlistFileExists() error {
 	if err := launchd.CheckPlist(*daemonConfig); err != nil {
 		return err
 	}
-	if !launchd.AgentRunning(daemonConfig.Label) {
+	_, err = daemonclient.GetVersionFromDaemonAPI()
+
+	if !launchd.AgentRunning(daemonConfig.Label) || err != nil {
```

Comment:
@anjannath conditions should be `if !launchd.AgentRunning(daemonConfig.Label) && err != nil` so that both need to be true before returning this error.

Reply:
Might be worth moving this to a small `func daemonRunning() bool` helper to make this more readable/less error-prone

Reply:
@praveenkumar @cfergeau updated